plugins/modules/system/parted.py

```diff
@@ -743,7 +736,7 @@ def main():
                 # Wrap double quotes in single quotes so the shell doesn't strip
                 # the double quotes as those need to be included in the arg
                 # passed to parted
-                script += 'name %s \'"%s"\' ' % (number, name)
+                script += ['name', str(number), name]
```

Comment:
```suggestion
                script += ['name', str(number), '"%s"' % name]
```

Reply:
I would love to hear from the module maintainers, or someone more familiar with `parted`: I cannot find anything in its documentation that suggests that adding an extra pair of quotes around any parameter is necessary. I think I will spin a VM up and run some tests to be sure.

Reply:
https://www.gnu.org/software/parted/manual/parted.html#name
It does not look like it *requires*  the quotes, but it suggests we should use them if needed (whitespace or something). As we move to the list form in `run_command`, I  reckon it is likely that's not necessary.

Reply:
Can you prove that all versions of parted this module supported in the past will work fine without the quotes, even if there are whitespaces in the name? Maybe parted itself just joins all arguments by space and then parses them; in that case the quotes would be absolutely necessary. (Also if your argument is correct, then we would have never needed the double quotes that were in the old version of the module.)

Reply:
I suspect indeed we never needed the double quotes to begin with, but not planning on spending time investigating that.

As of "all versions of parted this module supported in the past", if you can tell me which one is the earliest version this module claimed to support, I can try to take a look at parted code for that one. For the record, I am looking at the CLI parsing code in their git repo now, except for a json-output option that was added 4 months ago, the rest of the git blame is from: 8 years ago, 15 years ago, 12 years ago...

I will try and pinpoint the exact code that parses that, see how long it's been there for.

Reply:
> As of "all versions of parted this module supported in the past", if you can tell me which one is the earliest version this module claimed to support, I can try to take a look at parted code for that one.

I have no idea. Probably quite a few versions :)

Reply:
But considering that `parted /dev/sda help "print version"` and `parted /dev/sda help print version` both do the same, I'm *pretty sure* that the quotes are actually required so that spaces are handled correctly.

Reply:
That argument is meant to be a name provided by the user and as such, it may contain spaces. But yes, I see what you mean: `parted` tries to split each argument. Even the terminology used (_script_) for the trailing parameters indicates that. So I guess you are right.